### PR TITLE
feat: add toPipe and fromPipe commands

### PR DESCRIPTION
**Pull request by polvalente**
Adds support for the commands defined in https://github.com/elixir-lsp/elixir-ls/pull/515, keeping the fixes in https://github.com/elixir-lsp/elixir-ls/pull/520 in mind

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -408,7 +416,15 @@
         },
         {
           "command": "extension.expandMacro",
-          "title": "editorLangId == elixir"
+          "when": "editorLangId == elixir || editorLangId == eex || editorLangId == html-eex"
```
Comment (polvalente) [inline package.json:419]: Looking at line 415, this seemed to need a fix

**Comment by polvalente** [comment]
@axelson @lukaszsamson any tips on how to test this locally with the unreleased changes in ElixirLS? Should I just compile the LS and copy the files over to the local extension build for said test?

**Comment by lukaszsamson** [comment]
@polvalente I think there was a section in the readme some time ago. Maybe it was in the original Jake's repo. Open this repo in vscode and run debug with launch profile: Launch extension. It should build and release elixir-LS from submodule and launch a new debug vscode window.

**Comment by axelson** [comment]
> Should I just compile the LS and copy the files over to the local extension build for said test?

This is what I usually do. It looks something like `mix elixir_ls.release -o ~/.vscode-oss/extensions/elixir-lsp.elixir-ls-0.6.2/elixir-ls-release` (although I'm using the open source version of VSCode).

**Comment by polvalente** [comment]
> > Should I just compile the LS and copy the files over to the local extension build for said test?
> 
> This is what I usually do. It looks something like `mix elixir_ls.release -o ~/.vscode-oss/extensions/elixir-lsp.elixir-ls-0.6.2/elixir-ls-release` (although I'm using the open source version of VSCode).

Thanks! I've already tested and opened another fix on elixir-ls (https://github.com/elixir-lsp/elixir-ls/pull/521). It seems everything is working as expected, but there is a problem with sigils when there are `\` characters present (more details in the elixir-ls PR)

File: src/extension.ts [hunk 2]
```diff
@@ -211,7 +211,68 @@ function configureExpandMacro(context: ExtensionContext) {
...
+  const commandName = `extension.${operation}`;
+
+  const disposable = vscode.commands.registerCommand(commandName, async () => {
+    const extension = vscode.extensions.getExtension("jakebecker.elixir-ls");
+    const editor = vscode.window.activeTextEditor;
+    if (!extension || !editor) {
+      return;
+    }
+
+    const uri = editor.document.uri;
```
Comment (lukaszsamson) [inline src/extension.ts:243]: This client retrieval is a common part of custom commands and could be extracted into a function

File: src/extension.ts [hunk 3]
```diff
@@ -211,7 +211,68 @@ function configureExpandMacro(context: ExtensionContext) {
...
+
+interface ManipulatePipesResponseEdit {
+  newText: string,
+  range: {
+    start: {line: number, character: number},
+    end: {line: number, character: number}
+  }
+}
+
+interface ManipulatePipesResponse {
```
Comment (lukaszsamson) [inline src/extension.ts:226]: interface not used

**Comment by polvalente** [comment]
@lukaszsamson Sorry this took so long. PR is now ready for review again. One caveat is that ElixirLS needs to be updated in the submodule for this to work properly. I haven't committed this in this PR, but it'll be a necessary fix before releasing this feature :)

File: src/extension.ts [hunk 4]
```diff
@@ -201,7 +201,44 @@ function configureExpandMacro(context: ExtensionContext) {
...
+  context.subscriptions.push(disposable);
+}
+
+function configureManipulatePipes(context: ExtensionContext, operation: "toPipe" | "fromPipe") {
+  const commandName = `extension.${operation}`;
+
+  const disposable = vscode.commands.registerCommand(commandName, async () => {
+    const extension = vscode.extensions.getExtension("jakebecker.elixir-ls");
+    const editor = vscode.window.activeTextEditor;
+    if (!extension || !editor) {
```
Comment (lukaszsamson) [inline src/extension.ts:214]: Is the extension check necessary?

**Review by lukaszsamson — APPROVED** [review]

File: src/extension.ts (hunk 4 above)
Comment (polvalente) [reply src/extension.ts:214]: Not exactly, this just fails faster upon a bizarre error, if that makes sense. I can remove it if you prefer

**Comment by lukaszsamson** [comment]
@polvalente can you resolve merge conflicts?

**Comment by polvalente** [comment]
@lukaszsamson done. I did it through the web interface, so hopefully I didn't break the code. Please @ me if there are any problems!

**Review by lukaszsamson — APPROVED** [review]

**Pull request merged**